FWCore/Services/plugins/Timing.cc

```diff
@@ -143,9 +175,24 @@ namespace edm {
       iRegistry.watchPreEvent(this, &Timing::preEvent);
       iRegistry.watchPostEvent(this, &Timing::postEvent);
 
-      if(not summary_only_) {
+      if( (not summary_only_) || (threshold_ > 0.) ) {
         iRegistry.watchPreModuleEvent(this, &Timing::preModule);
         iRegistry.watchPostModuleEvent(this, &Timing::postModule);
+ 
```

Comment:
These can be in a different block using just` if( threshold_ > 0.)` since we do not want them in the case of `summary_only_ == false` but `threshold_ == 0.`.